presto-hive/src/main/java/io/prestosql/plugin/hive/HdfsEnvironment.java

```diff
@@ -128,8 +125,7 @@ public HdfsContext(ConnectorSession session, String schemaName, String tableName
             requireNonNull(schemaName, "schemaName is null");
             requireNonNull(tableName, "tableName is null");
             this.identity = requireNonNull(session.getIdentity(), "session.getIdentity() is null");
-            this.source = requireNonNull(session.getSource(), "session.getSource()");
-            this.queryId = Optional.of(session.getQueryId());
+            this.session = Optional.of(requireNonNull(session));
```

Comment:
Also drop identity and delagate to session.
Consider leaving the non-null checks though.
Add description to non-null check for session.

Reply:
> Also drop identity and delagate to session.

Identity cannot be dropped as session object is not always available (e.g some `GlueHiveMetastore` uses `DEFAULT_METASTORE_USER`). In the future (once we have Glue impersonation) maybe identity can be dropped

Reply:
Yeah - right.
I also noticed that there is `rnn` check for session already above. Drop the new one you added.